index.js

```diff
@@ -87,22 +94,23 @@ module.exports = function(opts) {
 				}
 
 				// listening port is chosen by NodeJS if opts.port === 0
-				const port = server.address().port;
-
-				broker.healthcheck.port = port;
-				broker.healthcheck.emit('port', port);
+				broker.healthcheck.port = server.address().port;
+				broker.healthcheck.emit('port', broker.healthcheck.port);
 
 				broker.logger.info('');
 				broker.logger.info('K8s health-check server listening on');
-				broker.logger.info(`    http://localhost:${port}${opts.readiness.path}`);
-				broker.logger.info(`    http://localhost:${port}${opts.liveness.path}`);
+				broker.logger.info(`    http://localhost:${broker.healthcheck.port}${opts.readiness.path}`);
+				broker.logger.info(`    http://localhost:${broker.healthcheck.port}${opts.liveness.path}`);
 				broker.logger.info('');
 			});
 		},
 
 		// After broker started
 		started(broker) {
 			state = 'up';
+
+			initChecker(checkersInfo, opts.readiness.path, opts.readiness.createChecker(broker), opts.readiness.timeoutMs);
```

Comment:
```js
[opts.readiness, opts.liveness].forEach((probe) =>
  initChecker(checkersMap, probe.path, probe.createChecker(broker), probe.timeoutMs)
);
```

The probe details can also be passed directly to `initChecker`:

```js
[opts.readiness, opts.liveness].forEach((probe) => initChecker(checkersMap, probe));
```

Reply:
Fixed in cddfcf6